lib/rpc/modules/admin.js

```diff
@@ -0,0 +1,56 @@
+'use strict'
+
+const util = require('ethereumjs-util')
+const { middleware } = require('../validation')
+const getClientVersion = require('../../util/getClientVersion')
+
+/**
+ * admin_* RPC module
+ * @memberof module:rpc/modules
+ */
+class Admin {
+  /**
+   * Create net_* RPC module
+   * @param {Node} Node to which the module binds
+   */
+  constructor (node) {
+    const service = node.services.find(s => s.name === 'eth')
+    this._chain = service.chain
+    this._node = node
+    this._ethProtocol = service.protocols.find(p => p.name === 'eth')
+
+    this.getNodeInfo = middleware(this.getNodeInfo.bind(this), 0, [])
+  }
+
+  /**
+   * Returns information about the currently running node.
+   * @param {*} [params] An empty array
+   * @param {*} [cb] A function with an error object as the first argument and the
+   */
+  async getNodeInfo (params, cb) {
+    const rlpxInfo = this._node.servers.find(s => s.name === 'rlpx').getRlpxInfo()
+    const name = getClientVersion()
+    const ethVersion = Math.max.apply(Math, this._ethProtocol.versions)
+    const network = this._chain.networkId
+    const lastetHeader = this._chain._headers.latest
+    const headHash = util.bufferToHex(lastetHeader.mixHash)
+    const difficutly = lastetHeader.difficulty
+    const genesisHash = util.bufferToHex(this._chain.genesis.hash)
+    const nodeInfo = {
+      ...rlpxInfo,
+      name,
+      protocols: {
+        eth: {
+          version: `eth/${ethVersion}`,
+          difficutly,
```

Comment:
Another deviation from the format that `Web3.py` delivers is that `difficulty` is presented in the following form as a buffer:

```json
"difficutly": {
                    "data": [
                        7,
                        111,
                        188,
                        163,
                        21,
                        132
                    ],
                    "type": "Buffer"
                }
```

Is this intended? Allowed?

I generally wonder: is there no written down specification for this RPC method, since it is not written down in the [JSON-RPC spec](https://github.com/ethereum/wiki/wiki/JSON-RPC)?

Reply:
yeah looks like we should convert difficulty to a number here.

here is geth's docs on the endpoint: https://geth.ethereum.org/docs/rpc/ns-admin#admin_nodeinfo

@holgerd77 check this out https://github.com/ethereum/hive/blob/master/clients/go-ethereum/enode.sh

Reply:
I think we should convert buffer to number.